webapp/src/containers/WalletPage/components/ReceivePage/CreateNewAddressPage/index.tsx

```diff
@@ -94,19 +100,22 @@ const CreateNewAddressPage: React.FunctionComponent<CreateNewAddressPageProps> =
   };
 
   const isAddressValid = async (value) => {
-    let isAddressValid = false;
-    if (
-      value.length >= 26 && // address, is an identifier of 26-35 alphanumeric characters
-      value.length <= 35
-    ) {
-      isAddressValid =
-        (await isValidAddress(value)) &&
-        (await isAddressMine(value)) &&
-        isAddressAlreadyExists(value);
-      // No need to check address belongs to HD wallet
-      // && (await hdWalletCheck(value));
+    if (value.length < 26 || value.length > 35) {
+      setAddressValidity(AddressValidity.ERROR_INVALID_ADDRESS_FORMAT);
+      return;
+    }
+    if (await isValidAddress(value)) {
+      setAddressValidity(AddressValidity.ERROR_INVALID_ADDRESS_FORMAT);
+      return;
+    }
+    if ((await isAddressMine(value)) && isAddressAlreadyExists(value)) {
+      setAddressValidity(AddressValidity.ERROR_EXIST_OWN_WALLET);
+      return;
+    }
+    if (await hdWalletCheck(value)) {
```

Comment:
I know it's WIP but don't do HD wallet check here it's commented for a reason

Reply:
Yup, just finding the right method to check the address existing check. It's not included on the final PR.